resource-definitions/template-driver/servicemonitor/servicemonitor-workload-def.yaml

```diff
@@ -0,0 +1,41 @@
+# This Resource Definition adds a Prometheus ServiceMonitor to a Workload
+apiVersion: entity.humanitec.io/v1b1
+kind: Definition
+metadata:
+  id: servicemonitor-workload
+entity:
+  driver_type: humanitec/template
+  name: servicemonitor-workload
+  type: workload
+  driver_inputs:
+    values:
+      namespace: ${resources['k8s-namespace#k8s-namespace'].outputs.namespace}
+      templates:
+        init: |
+          workload: {{ index (splitList "." "${context.res.id}") 1 }}
+          {{- if and .resource.spec.service (gt (.resource.spec.service.ports | len) 0) }}

```

Comment:
This will actually still fail if `.resource.spec.service` does not exist. This is because in Go Templates, there is no short circuit for and - both inputs are evaluated.

I would suggest something like:
```
          {{- if gt (.resource.spec | dig "service" "ports" (list)) 0 }}
```
See `dig` function documentation: https://masterminds.github.io/sprig/dicts.html#dig

Reply:
Oh ok, weird, because I tested w/ and w/o `service`, and `{{- if and .resource.spec.service (gt (.resource.spec.service.ports | len) 0) }}` worked.

Anyway, I just tested with `{{- if gt (.resource.spec | dig "service" "ports" (list) | len) 0 }}`, it works too, so taking this one based on your recommendation.